File: app/src/main/java/com/example/weizhenbin/wangebug/modules/collect/adapters/CollectListAdapter.java
package com.example.weizhenbin.wangebug.modules.collect.adapters;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Build;
import android.support.annotation.Nullable;
import android.text.Html;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.weizhenbin.wangebug.R;
import com.example.weizhenbin.wangebug.base.BaseSimpleAdapter;
import com.example.weizhenbin.wangebug.base.UrlTypeEnum;
import com.example.weizhenbin.wangebug.base.WebActivity;
import com.example.weizhenbin.wangebug.eventbus.EventBusHandler;
import com.example.weizhenbin.wangebug.eventbus.EventCode;
import com.example.weizhenbin.wangebug.eventbus.MessageEvent;
import com.example.weizhenbin.wangebug.modules.collect.controllers.CollectController;
import com.example.weizhenbin.wangebug.modules.collect.entity.TBCollectBean;
import com.example.weizhenbin.wangebug.tools.DialogTool;
import com.example.weizhenbin.wangebug.views.remindbar.RemindBar;

import java.util.List;

import static android.content.Context.CLIPBOARD_SERVICE;
import static android.text.Html.FROM_HTML_MODE_COMPACT;
import static com.example.weizhenbin.wangebug.views.remindbar.RemindBar.LENGTH_SHORT;

/**
 * Created by weizhenbin on 2018/10/10.
 */
public class CollectListAdapter extends BaseSimpleAdapter<TBCollectBean,BaseViewHolder> {
    public CollectListAdapter(@Nullable final Context context, @Nullable final List<TBCollectBean> data) {
        super(context, R.layout.collect_list_item, data);
        setOnItemChildClickListener(new OnItemChildClickListener() {
            @Override
            public void onItemChildClick(BaseQuickAdapter adapter, View view, int position) {
                if (data==null){
                    return;
                }
                TBCollectBean datasBean=   data.get(position);
                WebActivity.startActivity(context,datasBean.getUrl(),datasBean.getTitle(), UrlTypeEnum.valueOf(datasBean.getType()) );
            }
        });
        setOnItemChildLongClickListener(new OnItemChildLongClickListener() {
            @Override
            public boolean onItemChildLongClick(BaseQuickAdapter adapter, final View view, final int position) {
                String[] items=new String[]{mContext.getString(R.string.del_string),mContext.getString(R.string.copy_link_string)};
                DialogTool.showListAlertDialog(mContext, items, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (data==null){
                            return;
                        }
                        TBCollectBean tbCollectBean=data.get(position);
                        switch (which){
                            case 0:
                                CollectController.removeCollectByTitle(tbCollectBean.getTitle());
                                EventBusHandler.post(new MessageEvent(EventCode.CANCEL_COLLECT_CODE,tbCollectBean.getTitle()));
                                break;
                            case 1:
                                copyLike(view,tbCollectBean.getUrl());
                                break;
                        }
                    }
                });
                return true;
            }
        });
    }

    private void copyLike(View v,String url) {
        ClipboardManager clipboardManager=(ClipboardManager)mContext.getSystemService(CLIPBOARD_SERVICE);
        ClipData clipData = ClipData.newPlainText("text", url);
        if (clipboardManager!=null) {
            clipboardManager.setPrimaryClip(clipData);
        }
        RemindBar.make(v,mContext.getString(R.string.copy_success_string),LENGTH_SHORT).show();
    }

    @Override
    protected void convert(BaseViewHolder helper, TBCollectBean item) {
        if (Build.VERSION.SDK_INT>=Build.VERSION_CODES.N){
            helper.setText(R.id.tv_title, Html.fromHtml(item.getTitle(),FROM_HTML_MODE_COMPACT));
            helper.setText(R.id.tv_desc, Html.fromHtml(item.getUrl(),FROM_HTML_MODE_COMPACT));
        }else {
            helper.setText(R.id.tv_title,Html.fromHtml(item.getTitle()));
            helper.setText(R.id.tv_desc, Html.fromHtml(item.getUrl()));
        }
        helper.addOnClickListener(R.id.ll_item);
        helper.addOnLongClickListener(R.id.ll_item);
    }
}

File: app/src/main/java/com/example/weizhenbin/wangebug/modules/collect/adapters/CollectListAdapter.kt
package com.example.weizhenbin.wangebug.modules.collect.adapters

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Context.CLIPBOARD_SERVICE
import android.os.Build
import android.text.Html
import android.text.Html.FROM_HTML_MODE_COMPACT
import android.view.View
import com.chad.library.adapter.base.BaseQuickAdapter.OnItemChildClickListener
import com.chad.library.adapter.base.BaseQuickAdapter.OnItemChildLongClickListener
import com.chad.library.adapter.base.BaseViewHolder
import com.example.weizhenbin.wangebug.R
import com.example.weizhenbin.wangebug.base.BaseSimpleAdapter
import com.example.weizhenbin.wangebug.base.UrlTypeEnum
import com.example.weizhenbin.wangebug.base.WebActivity
import com.example.weizhenbin.wangebug.eventbus.EventBusHandler
import com.example.weizhenbin.wangebug.eventbus.EventCode
import com.example.weizhenbin.wangebug.eventbus.MessageEvent
import com.example.weizhenbin.wangebug.modules.collect.controllers.CollectController
import com.example.weizhenbin.wangebug.modules.collect.entity.TBCollectBean
import com.example.weizhenbin.wangebug.views.ListShortcutActionLayout
import com.example.weizhenbin.wangebug.views.remindbar.RemindBar
import com.example.weizhenbin.wangebug.views.remindbar.RemindBar.Companion.LENGTH_SHORT

/**
 * Created by weizhenbin on 2018/10/10.
 */
class CollectListAdapter(context: Context?, data: List<TBCollectBean>?) : BaseSimpleAdapter<TBCollectBean, BaseViewHolder>(context, R.layout.collect_list_item, data) {
    init {
        onItemChildClickListener = OnItemChildClickListener { _, _, position ->
            if (data == null) {
                return@OnItemChildClickListener
            }
            val datasBean = data[position]
            datasBean.url?.let {

                WebActivity.startActivity(context, it, datasBean.title, UrlTypeEnum.getValueFromType(datasBean.type))
            }
        }
        onItemChildLongClickListener = OnItemChildLongClickListener { _, view, position ->
            val items = arrayOf(mContext.getString(R.string.del_string), mContext.getString(R.string.copy_link_string))

            ListShortcutActionLayout.Builder(mContext).setAnchor(view).setItems(items).setiItemListener { which ->
                if (data == null) {
                    return@setiItemListener
                }
                val tbCollectBean = data[position]
                when (which) {
                    0 -> {
                        tbCollectBean.title?.let {
                            CollectController.removeCollectByTitle(it)
                            EventBusHandler.post(MessageEvent(EventCode.CANCEL_COLLECT_CODE, it))
                        }

                    }
                    1 -> {
                        tbCollectBean.url?.let {
                            copyLike(view,it)
                        }
                    }
                }
            }.build().show()
            true
        }
    }

    private fun copyLike(v: View, url: String) {
        val clipboardManager = mContext.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
        val clipData = ClipData.newPlainText("text", url)
            clipboardManager.primaryClip = clipData
        RemindBar.make(v, mContext.getString(R.string.copy_success_string), LENGTH_SHORT)?.show()
    }

    override fun convert(helper: BaseViewHolder, item: TBCollectBean) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            helper.setText(R.id.tv_title, Html.fromHtml(item.title, FROM_HTML_MODE_COMPACT))
            helper.setText(R.id.tv_desc, Html.fromHtml(item.url, FROM_HTML_MODE_COMPACT))
        } else {
            helper.setText(R.id.tv_title, Html.fromHtml(item.title))
            helper.setText(R.id.tv_desc, Html.fromHtml(item.url))
        }
        helper.addOnClickListener(R.id.ll_item)
        helper.addOnLongClickListener(R.id.ll_item)
    }
}
